app/components/tooltip/tooltip.tsx

```diff
@@ -2,9 +2,14 @@ import React from "react";
 import ReactDOM from "react-dom";
 import { clamp } from "../../util/math";
 
+export type MouseCoords = {
+  readonly clientX: number;
+  readonly clientY: number;
+};
+
 type TooltipProps = JSX.IntrinsicElements["div"] & {
   /** Renders the tooltip content if the tooltip is hovered. */
-  renderContent: () => React.ReactNode;
+  renderContent: (c?: MouseCoords) => React.ReactNode;
```

Comment:
Can this be `c: MouseCoords` so that users of the component don't need to handle undefined? (maybe if MouseCoords are unknown we can set to the last known value, or don't render initially until mouse coords are known?)

Reply:
Already more or less works as you described it: `renderContent` is not invoked unless the tooltip is set to `visible`, and the MouseCoords should be correct in that case.